tests/test_run_examples.py

```diff
@@ -54,7 +54,7 @@ def test_find_run_examples(example: CodeExample, eval_example: EvalExample):
 
     # assert 'my_file_9_13.py:12: AssertionError' in '\n'.join(result.outlines)
     assert result.outlines[-8:-3] == [
-        '',
+        '    b = 2',
```

Comment:
This is caused by changing to `f_code.co_firstlineno`, which seems to help pytest include more context from the original markdown file.

Reply:
I understand this properly now. Previously, setting co_firstlineno to `frame.f_lineno + example.start_line` was saying that the code started at the problematic line (ignoring the nuance about f_lineno vs tb_lineno) so pytest would show the failing line (so things appeared to work) but it thought that was the start of the code so it didn't show any context before that.

Leaving `co_firstlineno` unchanged meant it included more context, but too much.

`f_code.co_firstlineno + example.start_line` works correctly as you'd expect. That's the only way that this expanded test passes, with pytest showing the exact correct context.

Reply:
https://github.com/pydantic/pytest-examples/pull/19 doesn't pass this test because the code in `line_adjusted_source` still starts at the first line, even if it's just padding comments, so the full example file gets included as context for the top level frame.